### PR TITLE
Update testsuite

### PR DESCRIPTION
**Pull request by waterkip**
Word of warning: This PR contains changes I've submitted via e-mail to @timlegge and still needs to be reviewed. Code from the *last* commit of this PR should be reviewed and discussed.

****
In an older branch of mine I had some updates on the testsuite to make
it more useful. It mainly tries to move away from `ok($thing->func)` or
`ok('foo' eq 'bar')`.  syntax and actually testing the outcome of
whatever was called. In addition there are some testsuite helpers.

There are a couple of things that are different between .15/.17 and .25.

The tests in t/07-logout-request.t proved to be valuable, in my previous
branch everything worked. Some issues occured:

4e3ff81d4f2b963a855a5c24c96c87e2d4577c5b - this commit wipes the
NameIDPolicy in the XML from Net::SAML2::Protocol::AuthnRequest
(nameid_format attribute is now simply ignored).

7f459c303c250c532d5b47f8e137bdd88c4fc2a7 - this commit reintroduces it,
but under a different name: nameidpolicy_format. Now older code will
need to deal with this change. I therefore introduced an `around
BUILDARGS` which changes `nameid_format` to `nameidpolicy_format`.
Although I think we should pick *one* and stick to it. Other code uses
`nameid_format`: Net::SAML2::Protocol::LogoutRequest, Net::SAML2::SP and
Net::SAML2.

AllowCreate was completely ignored, this used to be true by default, but
now has an attribute with a predicate. If you set it, we'll use it,
otherwise it is ignored.

There are no other code changes in this PR, other than in the testsuite.
Hopefully this will make it easier to refactor code in the future.

Signed-off-by: Wesley Schwengle <wesley@opndev.io>

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
```
Comment (timlegge) [inline .editorconfig:1]: please add this file to .gitignore and force a squashed update

**Comment by waterkip** [comment]
> The Dockefile change looks fine too I need to look at how it interacts with the github action and whether it is complementary

It runs on cirrus ci from what I see, so that's a plus :)

**Comment by waterkip** [comment]
I think it is important to note that the commits introduce a breaking change (or rather the previous commits introduce breakage). Consumers should be made aware of this, I'm not certain what will go wrong downstream with the AllowCreate option. Now it is absent, false or true where is used to be true all the time.

**Comment by timlegge** [comment]
Tested successfully against my reference application Foswiki

**Pull request merged**